scheds/rust/scx_mitosis/src/bpf/mitosis.bpf.c

```diff
@@ -1264,6 +1301,52 @@ s32 BPF_STRUCT_OPS(mitosis_cgroup_init, struct cgroup *cgrp,
 	return 0;
 }
 
+/*
+ * Initialize cgroup and all its ancestors. Handles dying cgroups gracefully.
+ * Used when CPU controller is disabled since SCX cgroup callbacks won't fire.
+ */
+static int init_cgrp_ctx_with_ancestors(struct cgroup *cgrp)
+{
+	u32 target_level = cgrp->level;
+	u32 level;
+	int ret;
+
+	/* Skip dying cgroups */
+	if (cgrp_is_dying(cgrp))
+		return 0;
+
+	/* Initialize ancestors first (replicates SCX cgroup_init order) */
+	bpf_for(level, 1, target_level)
+	{
+		struct cgroup *ancestor = lookup_cgrp_ancestor(cgrp, level);
+		if (!ancestor)
+			return -ENOENT;
+
+		/* Skip if already initialized or dying */
+		if (!lookup_cgrp_ctx_fallible(ancestor) &&
+		    !cgrp_is_dying(ancestor)) {
```

Comment:
I'd flip these two conditionals since if its dying you'd be able to avoid the lookup